src/dmd/dtoh.d

```diff
@@ -2682,17 +2640,38 @@ public:
             printf("[AST.StructLiteralExp enter] %s\n", sle.toChars());
             scope(exit) printf("[AST.StructLiteralExp exit] %s\n", sle.toChars());
         }
+        const isUnion = sle.sd.isUnionDeclaration();
         sle.sd.type.accept(this);
         buf.writeByte('(');
         foreach(i, e; *sle.elements)
         {
             if (i)
                 buf.writestring(", ");
-            printExpressionFor(sle.sd.fields[i].type, e);
+
+            auto vd = sle.sd.fields[i];
+
+            // Expression may be null for unspecified elements
+            if (!e)
+                e = findDefaultInitializer(vd);
+
+            printExpressionFor(vd.type, e);
+
+            // Only emit the initializer of the first union member
+            if (isUnion)
+                break;
         }
         buf.writeByte(')');
     }
 
+    /// Finds the default initializer for the given VarDeclaration
+    private static AST.Expression findDefaultInitializer(AST.VarDeclaration vd)
```

Comment:
Likewise the introduction of this function into another PR.

Reply:
This change is already proposed in #12643, i just had to include it here s.t. dmd doesn't crash.